Game/Source/GDKTestGyms/GDKTestGymsGameInstance.cpp

```diff
@@ -21,19 +21,42 @@ void UGDKTestGymsGameInstance::OnStart()
 	}
 }
 
-bool UGDKTestGymsGameInstance::Tick(float DeltaSeconds)
+float UGDKTestGymsGameInstance::AddAndCalcFps(int64 NowReal, float DeltaS)
 {
-	float Alpha = 0.8f;
-	float CurrFPS = 1.0f / DeltaSeconds; // TODO - we should remove FApp::GetIdleTime() from DeltaSeconds once the NFR regex is fixed
-	AverageFPS = (Alpha * AverageFPS) + ((1.0f - Alpha) * CurrFPS);
+	TicksForFPS.Push(TPair<int64, float>(NowReal, DeltaS)); 
+	float TotalDelta = 0.0f;
+	int NumSamples = 0;
+	int64 MinRange = NowReal - FTimespan::FromMinutes(2.0).GetTicks();
+	for (int i = TicksForFPS.Num()-1; i >= 0; i--)
+	{
+		const FPSTimePoint& TimePoint = TicksForFPS[i];
+		if (TimePoint.Key < MinRange)
+		{
+			TicksForFPS.RemoveAt(i);
+		}
+		else
+		{
+			TotalDelta += TimePoint.Value;
+			NumSamples++;
+		}
+	}
+	if (NumSamples > 0)
```

Comment:
Could we not just use TicksForFPS.Num() here. Do we need to increment NumSamples in the loop?

Reply:
Can we calculate the "TotalDelta" from the Datetime ticks? 
We could look at the first element and last element of TicksForFPS, and perhaps use their "NowReal" times to calculate the DeltaTime?

Reply:
This might also allow us to break early from the loop once we have removed all elements who fall outside the window (if TicksForFPS is ordered which I guess it is)

Reply:
If I have explained this poorly we can have a chat :)

Reply:
`Do we need to increment NumSamples in the loop?` Good point can just use the array size
`Can we calculate the "TotalDelta" from the Datetime ticks?` Good point we possibly could, although we use DeltaTime for the Spatial metrics too and this doesn't always match real time (max delta capping). It's worth thinking about performance though because the array length is ~7200 so it could be worth thinking about how we can accumulate better - maybe using the sliding window method of adding+subtracting when a value is added/removed (but without float to avoid precision loss over long periods)

Reply:
Yeah maybe having a member variable that gets added to whenever we get a new DeltaSeconds. We also subtract any TimePoints deltas that were not in the 2 min window. This allows us to only loop over a few elements and break early.